cmake/HPX_AddComponent.cmake

```diff
@@ -102,18 +102,34 @@ function(add_hpx_component name)
     set(exclude_from_all EXCLUDE_FROM_ALL)
   else()
     if(${name}_PLUGIN AND NOT HPX_WITH_STATIC_LINKING)
-      set(install_destination ${CMAKE_INSTALL_LIBDIR}/hpx)
+      if(MSVC)
+        set(library_install_destination ${CMAKE_INSTALL_BINDIR}/hpx)
+      else()
+        set(library_install_destination ${CMAKE_INSTALL_LIBDIR}/hpx)
+      endif()
+      set(archive_install_destination ${CMAKE_INSTALL_LIBDIR}/hpx)
+      set(runtime_install_destination ${CMAKE_INSTALL_BINDIR}/hpx)
       set(${name}_OUTPUT_SUFFIX hpx)
     else()
-      set(install_destination ${CMAKE_INSTALL_LIBDIR})
+      if(MSVC)
+        set(library_install_destination ${CMAKE_INSTALL_BINDIR})
+      else()
+        set(library_install_destination ${CMAKE_INSTALL_LIBDIR})
+      endif()
+      set(archive_install_destination ${CMAKE_INSTALL_LIBDIR})
+      set(runtime_install_destination ${CMAKE_INSTALL_BINDIR})
     endif()
     if(${name}_INSTALL_SUFFIX)
-      set(install_destination ${${name}_INSTALL_SUFFIX})
+      set(library_install_destination ${${name}_INSTALL_SUFFIX})
+      set(archive_install_destination ${${name}_INSTALL_SUFFIX})
+      set(runtime_install_destination ${${name}_INSTALL_SUFFIX})
     endif()
     set(_target_flags
       INSTALL
       INSTALL_FLAGS
-        DESTINATION ${install_destination}
+        LIBRARY DESTINATION ${library_install_destination}
+        ARCHIVE DESTINATION ${archive_install_destination}
+        RUNTIME DESTINATION ${runtime_install_destination}
```

Comment:
Do we actually need `RUNTIME DESTINATION` here? Not that it hurts, but a component should never be an executable. Same in `HPX_AddLibrary.cmake`.

Reply:
Indeed. That can be removed.

Reply:
Correction: on Windows the `RUNTIME` path is required as its used to install the generated dll's.

Reply:
Ah, ok. Then it stays.